crates/transaction-pool/src/validate/eth.rs

```diff
@@ -728,4 +796,47 @@ mod tests {
         let tx = pool.get(transaction.hash());
         assert!(tx.is_some());
     }
+
+    #[cfg(feature = "optimism")]
+    #[tokio::test(flavor = "multi_thread")]
+    async fn validate_optimism_transaction() {
```

Comment:
we should add a bench for this if it does the i/o + recovery, otherwise the txpool feels like it'll be slow